### PR TITLE
feat: add implementation guide

### PR DESCRIPTION
**Pull request by cellomatt**
### Description

This PR 

- Adds a new implementation guide that outlines the steps for planning your Knock integration
- Links the new guide in the sidebar under the title "Plan your Knock integration" (please give feedback on that copy if you have any)
- Removes the "Going to production" guide and updates links that referenced it so they point to the similar section in this guide
- Creates/updates redirects for previous links to the guide that is being removed
- Updates the mAPI page under Developer tools to reference newly-released features that are available in mAPI

https://docs-git-mk-implementation-guide-knocklabs.vercel.app/guides/implementation-guide

### CODE REVIEW

File: data/sidebar.ts [hunk 1]
```diff
@@ -336,6 +336,10 @@ const sidebarContent: SidebarSection[] = [
     title: "Guides",
     slug: "/guides",
     pages: [
+      {
+        slug: "/implementation-guide",
+        title: "Plan your Knock integration",
```
Comment (samseely) [inline data/sidebar.ts:341]: @cellomatt I think this works but a couple other options:
* "Knock implementation guide" — just same as the title
* "Implementation guide" — if we want it to be shorter
* "Plan your Knock implementation" — same as you have but title of doc is implementation and then we have integration in sidebar

File: content/guides/implementation-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+## Going to production
+
+You’re finally ready to move to production – congratulations! Before you flip the switch and sit back to admire your work, here are a few last-minute items to check off your list:
+
+- Be sure to update your application’s [API keys](/developer-tools/api-keys) to point to your Knock Production environment.
+- Ensure that all of your work has been [promoted](/concepts/commits#promoting-commits) to your Production environment.
+- If you’re using in-app messaging, [generate a signing key](/in-app-ui/security-and-authentication#1-generate-the-signing-key) and enable enhanced security mode for client-side requests in your Production environment. Both of these actions can be completed by navigating to **Developers** > **API Keys** in your Knock dashboard’s Production environment.
+
+As always, we’re here to help. If you have any questions or run into issues as you build with our product, we hope you’ll let us know! Email us at support@knock.app and we’ll be more than happy to assist you. Knock on 🤘
```
Comment (samseely) [inline content/guides/implementation-guide.mdx:261]:
```suggestion
As always, we’re here to help. If you have any questions or run into issues as you build with our product, we hope you’ll let us know! Email us at support@knock.app and we’ll be more than happy to assist you. 

Knock on. 🤘
```

This is the ultimate nit but I kind of like a line break before that final sign off, but more a personal preference than anything so just go with whatever you prefer

**Review by samseely — APPROVED** [review]
🚀🚀🚀

File: data/sidebar.ts (hunk 1 above)
Comment (cellomatt) [reply data/sidebar.ts]: 💯

**Pull request merged**